File: frontend/src/types/emotion.ts
export type EmotionKey = "angry" | "disgust" | "fear" | "happy" | "sad" | "surprise" | "neutral";

export interface EmotionScores {
  angry: number;
  disgust: number;
  fear: number;
  happy: number;
  sad: number;
  surprise: number;
  neutral: number;
}

export interface BBox {
  x: number;
  y: number;
  w: number;
  h: number;
}

export interface EmotionRecord extends EmotionScores {
  id: string;
  frame_index: number;
  timestamp_ms: number;
  person_id: string;
  dominant_emotion: EmotionKey;
  intensity: number;
  bbox: BBox | null;
}

export interface TimelineBucket extends EmotionScores {
  timestamp_ms: number;
  intensity: number;
}

export interface EmotionTimeline {
  session_id: string;
  duration_ms: number;
  bucket_ms: number;
  persons: string[];
  series: TimelineBucket[];
}

export interface TrackedPerson {
  id: string;
  track_id: number;
  first_seen_ms: number;
  last_seen_ms: number;
  frame_count: number;
  dominant_emotion: EmotionKey | null;
  avg_intensity: number | null;
}

File: backend/app/schemas/emotion.py
from typing import Optional
from pydantic import BaseModel


class BBox(BaseModel):
    x: float
    y: float
    w: float
    h: float


class EmotionRecordResponse(BaseModel):
    id: str
    frame_index: int
    timestamp_ms: float
    person_id: str
    angry: float
    disgust: float
    fear: float
    happy: float
    sad: float
    surprise: float
    neutral: float
    dominant_emotion: str
    intensity: float
    bbox: Optional[BBox]

    model_config = {"from_attributes": True}


class TimelineBucket(BaseModel):
    timestamp_ms: float
    angry: float
    disgust: float
    fear: float
    happy: float
    sad: float
    surprise: float
    neutral: float
    intensity: float


class EmotionTimelineResponse(BaseModel):
    session_id: str
    duration_ms: float
    bucket_ms: int
    persons: list[str]
    series: list[TimelineBucket]


class PersonResponse(BaseModel):
    id: str
    track_id: int
    first_seen_ms: float
    last_seen_ms: float
    frame_count: int
    dominant_emotion: Optional[str]
    avg_intensity: Optional[float]

    model_config = {"from_attributes": True}
